docs/guides/monorepo-react.md

```diff
@@ -0,0 +1,177 @@
+
+## Generating a React Application
+
+Run
+
+```bash
+ng g app frontend --framework=react
+```
+
+and you will the following:
+
+```treeview
+myworkspace/
+├── README.md
+├── angular.json
+├── apps/
+│   ├── frontend/
+│   │   ├── src/
+│   │   │   ├── app/
+│   │   │   │   ├── app.css
+│   │   │   │   ├── app.spec.tsx
+│   │   │   │   └── app.tsx
+│   │   │   ├── assets/
+│   │   │   ├── environments/
+│   │   │   ├── favicon.ico
+│   │   │   ├── index.html
+│   │   │   ├── main.ts
+│   │   │   ├── polyfills.ts
+│   │   │   ├── styles.scss
+│   │   │   └── test.ts
+│   │   ├── browserslist
+│   │   ├── jest.conf.js
+│   │   ├── tsconfig.app.json
+│   │   ├── tsconfig.json
+│   │   ├── tsconfig.spec.json
+│   │   └── tslint.json
+│   └── frontend-e2e/
+│   │   ├── src/
+│   │   │   ├── integrations/
+│   │   │   │   ├── app.spec.ts
+│   │   │   ├── fixutres/
+│   │   │   ├── plugins/
+│   │   │   └── support/
+│   │   ├── cypress.json
+│   │   ├── tsconfig.e2e.json
+│   │   └── tslint.json
+├── libs/
+├── nx.json
+├── package.json
+├── tools/
+├── tsconfig.json
+└── tslint.json
+```
+
+Run:
+
+- `ng serve frontend` to serve the application
+- `ng build frontend` to build the application
+- `ng lint frontend` to lint the application
+- `ng test frontend` to test the application using Jest
+- `ng e2e frontend-e2e` to test the application using Cypress
+
+As with Angular and Node, when using React in Nx, you get the out-of-the-box support for TypeScript, Cypress, Jest. No need to configure anything: watch mode, source maps, and typings just work.
+
+## Generating a React Library
+
+Run
+
+```bash
+ng g lib home --framework=react
+```
+
+and you will the following:
+
+```treeview
+myworkspace/
+├── README.md
+├── angular.json
+├── apps/
+│   ├── frontend/
+│   └── frontend-e2e/
+├── libs/
+│   └── home/
+│       ├── src/
+│       │   ├── lib/
+│       │   │    ├── home.css
+│       │   │    ├── home.tsx
+│       │   │    └── home.spec.tsx
+│       │   └ index.ts
+│       ├── jest.config.js
+│       ├── tsconfig.json
+│       ├── tsconfig.lib.json
+│       ├── tsconfig.spec.json
+│       └── tslint.json
+├── nx.json
+├── package.json
+├── tools/
+├── tsconfig.json
+└── tslint.json
+```
+
+Run:
+
+- `ng test home` to test the library
+- `ng lint home` to lint the library
+
+## Using Library in Application
+
+You can import the home library into the frontend application like this.
+
+```typescript
+import { Component } from 'react';
+import { Home } from '@myworkspace/home';
+
+import './app.css';
+
+export class App extends Component {
+  render() {
+    return (
+      <>
+        <Home />
+        <div>rest of app component</div>
+      </>
+    );
+  }
+}
+```
+
+## Sharing Code
+
+Without Nx creating a new shared library can take weeks: a new repo needs to be provisioned, CI needs to be set up, etc.. In an Nx Workspace it takes minutes.
```

Comment:
I probably say "takes hours"? Weeks sound like a lot, especially for React "hackers".

Reply:
changed it from several hours to weeks

The main reason why it takes weeks is that setting a new repo, cli, publishing to artifactory may take hours to implement, but in most companies getting all the approvals in place will take weeks.